src/test/java/org/eolang/speco/MainTest.java

```diff
@@ -61,21 +70,52 @@ public final class MainTest {
      */
     private final Path eos = this.tests.resolve("eo");
 
+    @Tag("fast")
     @ParameterizedTest
     @ValueSource(strings = {"simple"})
     public void convertsFromXmir(final String name, @TempDir final Path temp) throws IOException {
         MainTest.compare(temp, MainTest.runSpeco(this.xmirs.resolve(name), temp, false));
     }
 
+    @Disabled
+    @Tag("fast")
     @ParameterizedTest
-    @ValueSource(strings = {"booms", "pets"})
+    @MethodSource("getEoTests")
     public void convertsFromEo(final String name, @TempDir final Path temp) throws IOException {
-        final Path base = this.eos.resolve(name);
-        MainTest.compare(temp, MainTest.runSpeco(base, temp, true));
+        MainTest.compare(temp, MainTest.runSpeco(this.eos.resolve(name), temp, true));
+    }
+
+    @Tag("slow")
+    @ParameterizedTest
+    @MethodSource("getEoTests")
+    public void compilesFromEo(final String name, @TempDir final Path temp) throws IOException {
+        MainTest.runSpeco(this.eos.resolve(name), temp, true);
         Assertions.assertEquals(
-            Files.readAllLines(base.resolve("result.txt")),
-            this.exec(temp.toString())
+            Files.readAllLines(this.eos.resolve(name).resolve("result.txt")),
+            this.exec(temp.toString()),
+            String.format("Program %s produced an incorrect result", name)
+        );
+    }
+
+    /**
+     * Generates full names of eo test cases.
+     *
+     * @return Collection of test cases names
+     */
+    @SuppressWarnings("PMD.UnusedPrivateMethod")
+    private static Collection<String> getEoTests() {
```

Comment:
@advasileva Maybe `cases()` would be better name? What do you think?
https://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html

Reply:
I agree, but this method is no longer needed and has been removed